File: ignite-nest/src/comment/infrastructure/repository/comment-command.adapter.ts
import { CommentOutPort } from '../../domain/port/out/comment-out.port';
import { Injectable } from '@nestjs/common';
import { Repository } from 'typeorm';
import { InjectRepository } from '@nestjs/typeorm';
import { Comment } from '../../entity/comment.entity';

@Injectable()
export class CommentCommandAdapter implements CommentOutPort {
  constructor(
    @InjectRepository(Comment)
    private readonly commentRepository: Repository<Comment>,
  ) {}

  async deleteByPostId(postId: number): Promise<void> {
    await this.commentRepository.delete({ postId });
  }

  async createComment(comment: Comment): Promise<Comment> {
    return await this.commentRepository.save(comment);
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/comment/application/adapter/CommentCommandAdapter.kt
package org.rookedsysc.ignitespring.comment.application.adapter

import org.rookedsysc.ignitespring.comment.domain.port.out.CommentOutPort
import org.rookedsysc.ignitespring.comment.infrastructure.repository.CommentRepository
import org.springframework.stereotype.Repository
import javax.xml.stream.events.Comment

@Repository
class CommentCommandAdapter(
    private val commentRepository: CommentRepository
): CommentOutPort {
    override fun createComment(comment: Comment): Comment {
        return commentRepository.save(comment)
    }
}
